fixtures/type-limits/tests/bindings/test_type_limits.py

```diff
@@ -195,11 +195,48 @@ def test_special_floats(self):
         self.assertTrue(math.isnan(take_f32(math.nan)))
         self.assertTrue(math.isnan(take_f64(math.nan)))
 
+    def test_non_string(self):
+        self.assertRaises(TypeError, lambda: take_string(None))
+        self.assertRaises(TypeError, lambda: take_string(False))
+        self.assertRaises(TypeError, lambda: take_string(True))
+        self.assertRaises(TypeError, lambda: take_string(0))
+        self.assertRaises(TypeError, lambda: take_string(0.0))
+        self.assertRaises(TypeError, lambda: take_string(b""))
+
+        class A:
```

Comment:
It's not clear to me what the current semantics are - does this work without this patch? If so, what's the motivation for preventing this from working?

Reply:
(I mean, I guess "does the object have `__str__`?" isn't really a great indicator of "this object wants to be seen as string like", but my question is really more about "do we want to exclude objects which want to be string like?"

Reply:
> It's not clear to me what the current semantics are - does this work without this patch?

I didn't change the current semantics except for throwing `TypeError` instead of `AttributeError`. So no, it does not work without this patch.

In this way, this test is trying to test for the current behavior.